### PR TITLE
Feauture/39038 csv import more fields

### PR DESCRIPTION
**Pull request by lucatume**
Ticket: https://central.tri.be/issues/39038

This PR adds the following supports to the CSV import:

* venue description (`post_content`)
* organizer description (`post_content`)

plus some `Tribe__Events__Importer__File_Importer` refactoring to allow for some method reuse

see https://github.com/moderntribe/events-pro/pull/303 for venue PRO supports

### CODE REVIEW

File: src/Tribe/Importer/Column_Mapper.php [hunk 1]
```diff
@@ -71,27 +71,29 @@ private function get_event_column_names() {
...
-			'venue_address'  => esc_html__( 'Venue Address', 'the-events-calendar' ),
-			'venue_address2' => esc_html__( 'Venue Address 2', 'the-events-calendar' ),
-			'venue_city'     => esc_html__( 'Venue City', 'the-events-calendar' ),
-			'venue_state'    => esc_html__( 'Venue State/Province', 'the-events-calendar' ),
-			'venue_zip'      => esc_html__( 'Venue Zip', 'the-events-calendar' ),
-			'venue_phone'    => esc_html__( 'Venue Phone', 'the-events-calendar' ),
-			'venue_url'      => esc_html__( 'Venue Website', 'the-events-calendar' ),
-			'featured_image' => esc_html__( 'Venue Featured Image', 'the-events-calendar' ),
-		);
+		return apply_filters( 'tribe_events_importer_venue_column_names', array(
```
Comment (barryhughes) [inline src/Tribe/Importer/Column_Mapper.php]: Nice! Can we add a docblock for the filter though?

File: src/Tribe/Importer/Column_Mapper.php [hunk 2]
```diff
@@ -71,27 +71,29 @@ private function get_event_column_names() {
...
+			'venue_state'       => esc_html__( 'Venue State/Province', 'the-events-calendar' ),
+			'venue_zip'         => esc_html__( 'Venue Zip', 'the-events-calendar' ),
+			'venue_phone'       => esc_html__( 'Venue Phone', 'the-events-calendar' ),
+			'venue_url'         => esc_html__( 'Venue Website', 'the-events-calendar' ),
+			'featured_image'    => esc_html__( 'Venue Featured Image', 'the-events-calendar' ),
+		) );
 	}
 
 	private function get_organizer_column_names() {
 		return array(
```
Comment (barryhughes) [inline src/Tribe/Importer/Column_Mapper.php]: Optionally, could be nice to make this filterable (like with `get_venue_column_names()`), too.

File: src/Tribe/Importer/File_Importer.php [hunk 3]
```diff
@@ -208,4 +208,25 @@ public function filter_query_for_title_search( $search, WP_Query $wp_query ) {
 	protected function featured_image_uploader( $featured_image ) {
 		return empty( $this->featured_image_uploader ) ? new Tribe__Events__Importer__Featured_Image_Uploader( $featured_image ) : $this->featured_image_uploader;
 	}
+
+	/**
+	 * Returns a boolean value from the record.
+	 *
+	 * @param        array  $record
```
Comment (barryhughes) [inline src/Tribe/Importer/File_Importer.php]: Some funky spacing between @param and the type on this line and the next.

File: src/Tribe/Importer/File_Importer.php [hunk 4]
```diff
@@ -208,4 +208,25 @@ public function filter_query_for_title_search( $search, WP_Query $wp_query ) {
...
+	 * @param        string $key
+	 * @param string        $return_true_value    The value to return if the value was found and is truthy.
```
Comment (barryhughes) [inline src/Tribe/Importer/File_Importer.php]: ...more funky spacing here too.

File: tests/functional/Tribe/Events/Importer/File_Importer_VenuesTest.php [hunk 5]
```diff
@@ -64,70 +73,30 @@ public function it_should_be_instantiatable() {
...
-		$image_url     = plugins_url( '_data/csv-import-test-files/featured-image/images/featured-image.jpg', codecept_data_dir() );
-		$attachment_id = $this->factory()->attachment->create_upload_object( $image_url );
-		$this->featured_image_uploader->upload_and_get_attachment()->willReturn( $attachment_id );
 
-		$sut = $this->make_instance();
+	protected function make_instance( $template_dir = null ) {
+		$this->setup_file( $template_dir );
 
-		$post_id = $sut->import_next_row();
+		$sut = new Venues_Importer( $this->file_reader, $this->featured_image_uploader->reveal() );
```
Comment (barryhughes) [inline tests/functional/Tribe/Events/Importer/File_Importer_VenuesTest.php]: What does `$sut` mean?

Comment (lucatume) [inline tests/functional/Tribe/Events/Importer/File_Importer_VenuesTest.php]: subject under test

Comment (barryhughes) [inline tests/functional/Tribe/Events/Importer/File_Importer_VenuesTest.php]: I was showing my ignorance, @borkweb clued me in there.

Comment (lucatume) [inline tests/functional/Tribe/Events/Importer/File_Importer_VenuesTest.php]: It's common to use a var name like that to have a clear representation that:

* you are actually testing **one** thing only
* what you are testing

**Comment by barryhughes** [comment]
:+1:

**Pull request merged**